### PR TITLE
Adapters manager implementation

### PR DESCRIPTION
**Pull request by PeterBolha**

### CODE REVIEW

File: adapters/AdaptersManager.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                return getattr(adapter_impl, method_name)(*args)
+            except AdapterSkipException:
+                self._logger.warning(
+                    f'Method "{method_name}" is not supported by '
+                    f'{current_adapter["name"]}. Going to try another '
+                    f'adapter if available.')
+                current_priority += 1
+                current_adapter = self._ADAPTERS.get(current_priority)
+            except ApiException as ex:
+                if 'NotExistsException"' in ex.body:
```
Comment (xpavlic) [inline adapters/AdaptersManager.py:69]:
```suggestion
                if 'notexistsexception"' in ex.body.lower():
```

**Review by xpavlic — CHANGES_REQUESTED** [review]

**Comment by xpavlic** [comment]
Also please rebase branche to resolve conflict and drop [Integrated correct adapters interface](https://github.com/CESNET/proxyaai_perun_connection_manager/pull/13/commits/95ba23f4dd4c70d104c3af46b2201c2f31830162)

File: adapters/AdaptersManager.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+                self._ADAPTERS[priority] = {
+                    "name": "rpc_adapter",
+                    "adapter": rpc_adapter,
+                }
+            else:
+                self._logger.warning(
+                    f'Config file includes unsupported adapter type "'
+                    f'{adapter_type}"'
+                )
```
[the remark this answers is not in the document]
Comment (PeterBolha) [reply adapters/AdaptersManager.py:51]: Okay so in this case it means I should remove the ConfigStore.get_xxx_config() and extract the information from the `adapter_info` instead?

[the remark this answers is not in the document]
Comment (xpavlic) [reply adapters/AdaptersManager.py:51]: Exactly.

File: adapters/AdaptersManager.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        return self.execute_method_by_priority(
+            self.get_caller_name(), idp_id, uids
+        )
+
+    def get_group_by_name(self, vo: VO, name: str) -> Group:
+        return self.execute_method_by_priority(
+            self.get_caller_name(), vo, name
+        )
+
+    def get_vo(self, short_name=None, id=None) -> VO:
```
Comment (xpavlic) [inline adapters/AdaptersManager.py:97]: I would add type (shor_name: str,  id: int)

**Review by xpavlic — CHANGES_REQUESTED** [review]

File: adapters/AdaptersManager.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+from models.Group import Group
+from models.User import User
+from models.UserExtSource import UserExtSource
+from models.VO import VO
+from perun_openapi import ApiException
+from utils.ConfigStore import ConfigStore
+
+
+class AdaptersManager(AdapterInterface):
+    def __init__(self):
```
Comment (xpavlic) [inline adapters/AdaptersManager.py:20]: Pass config as parameter

File: adapters/AdaptersManager.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+        self._logger = Logger.get_logger(self.__class__.__name__)
+        self._STARTING_PRIORITY = 1
+        self._ADAPTERS = {}
```
Comment (xpavlic) [inline adapters/AdaptersManager.py:23]: Remove underscore, we need to access it in tests.

**Review by xpavlic — CHANGES_REQUESTED** [review]

File: adapters/AdaptersManager.py (hunk 4 above)
Comment (PeterBolha) [reply adapters/AdaptersManager.py:20]:
```
def __init__(self, config=ConfigStore.get_adapters_manager_config()):
    self._logger = Logger.get_logger(self.__class__.__name__)
    self._STARTING_PRIORITY = 1
    self.adapters = {}

    adapters_info = config["adapters"]
```

Comment (xpavlic) [reply adapters/AdaptersManager.py:20]: exactly

File: adapters/AdaptersManager.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+                    "name": "rpc_adapter",
+                    "adapter": rpc_adapter,
+                }
+            else:
+                self._logger.warning(
+                    f'Config file includes unsupported adapter type "'
+                    f'{adapter_type}"'
+                )
+
+    def execute_method_by_priority(self, method_name: str, *args):
```
Comment (xpavlic) [inline adapters/AdaptersManager.py:52]: add _ to method name

File: adapters/AdaptersManager.py [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+                    f'successfully by {current_adapter["name"]}, exception '
+                    f'occurred: "{ex}"')
+                raise
+
+        raise Exception(
+            f'None of the provided adapters was able to resolve method "'
+            f'{method_name}"'
+        )
+
+    def get_caller_name(self):
```
Comment (xpavlic) [inline adapters/AdaptersManager.py:84]: same as above

**Review by xpavlic — CHANGES_REQUESTED** [review]
Squash commits, then we will merge this.

**Pull request merged**